peachjam/adapters/judgments.py

```diff
@@ -0,0 +1,309 @@
+import logging
+import math
+from mimetypes import guess_extension
+from tempfile import NamedTemporaryFile
+
+import magic
+import requests
+from cobalt.uri import FrbrUri
+from countries_plus.models import Country
+from django.core.files import File
+from django.utils import timezone
+from django.utils.text import slugify
+from languages_plus.models import Language
+
+from peachjam.adapters.base import RequestsAdapter
+from peachjam.models import (
+    CaseNumber,
+    Court,
+    CourtRegistry,
+    DocumentTopic,
+    Judge,
+    Judgment,
+    MatterType,
+    SourceFile,
+    Taxonomy,
+)
+from peachjam.plugins import plugins
+
+log = logging.getLogger(__name__)
+
+
+class BaseJudgmentAdapter(RequestsAdapter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.client.headers.update(
+            {
+                "Authorization": f"Token {self.settings['token']}",
+            }
+        )
+        self.filters = {}
+        for pair in (self.settings.get("filters") or "").split():
+            if "=" in pair:
+                key, value = pair.split("=")
+                self.filters[key] = value
+        self.court_codes = self.settings["court_code"].split()
+        self.filters["court__code"] = []
+        for code in self.court_codes:
+            self.filters["court__code"].append(code)
+
+    def client_request(self, method, url, **kwargs):
```

Comment:
this feels like it could either go on RequestsAdapter, or is repeating something already there

Reply:
done